Spotlight: Kurt Unger (#770)
OpenSats/website: data/blog/developer-spotlight-kurt-unger.mdx

---

> title: "Developer Spotlight: Kurt Unger"

Reviewer: ```suggestion
title: "Made in Africa: Kurt and the Open-Source Bitcoin Miner That Anyone Can Make"
```

date: '2026-05-22'
tags: ['OpenSats', 'bitcoin', 'hardware', 'spotlight']
authors: ['ville']
images: ['/static/images/spotlight/kurt-unger/featured.jpg']
draft: false

> summary: "A developer spotlight on Kurt Unger. Making Bitcoin mining hardware understandable, repairable, and open."

Reviewer 2: Summary is worth improving imho.

> summary: "A developer spotlight on Kurt Unger. Making Bitcoin mining hardware understandable, repairable, and open."

Reviewer: ```suggestion
summary: "A developer spotlight on Kurt Unger, an OpenSats grantee who's building Bitcoin mining hardware that's understandable, repairable, and open."
```

was tough.

During those years in Tanzania, the family grew by two (and later by three), and
Kurt's interest in renewable energy grew with it. He moved back to Canada to
study electrical engineering, then returned to Kenya to work on hardware for
emerging markets.

Eventually, that path led him to *Gridless*, the Nairobi-based company that uses
surplus renewable energy in rural Africa to mine bitcoin. There, Kurt applied
his knowledge and skills to maintain and repair the mining hardware. That's
where Bitcoin became less magic internet money and more infrastructural fact.

>*My own, smaller, a-ha moment came when I realized how truly open-source it
>\[Bitcoin\] was.*

The bigger picture clicked when he started thinking about microgrids. Small
hydro and solar projects across rural Africa share a common problem: the
communities around them need power, but not steadily enough to keep the project
viable. Surplus electricity goes unused.

That's when Bitcoin miners can take whatever power the village isn't using, and
then go quiet when the village needs the electricity. In doing so, the community
gets reliable electricity that it wouldn't otherwise have. Kurt remembers fondly

> that "it is quite a good feeling to see a small community productively using

Reviewer: From Kurt:  "My only comment is that I see you ended up using that one convoluted sentence 

> "it is quite a good feeling to see a small community productively using energy, knowing that it was partially enabled by the work I did to provide the baseload income which helped fund the microgrid."

Feel free to change it around and pretend like I quoted it! Or I can have another stab at it."

Reviewer 2: I think that sentence is perfectly fine as is!

energy, knowing that it was partially enabled by the work I did to provide the
baseload income which helped fund the microgrid."

And this is where [open hardware][open-hardware] starts to matter: if the miners
are expensive to import, hard to repair, and built by a handful of companies far
away, then the energy may be local, but the power still isn't.

[*Bitshoka Nini*][bitshoka-nini] starts where most open-source projects do:
building upon existing work. "Shoka" in Swahili means axe, hence the
[*Bitaxe*][bitaxe], an open-source bitcoin miner that was also funded by
OpenSats. In open source, copying is the point: someone publishes the files,
then somebody else picks them up, builds, breaks, and fixes them before putting
the work back into the wild.

The main difference between the two is the ASIC, or the application-specific
integrated circuit. An ASIC is a specialist in a world of generalists: useless
at nearly everything, but frighteningly good at one thing. Namely, guessing
numbers until you get the right one. However, no one had publicly analyzed the

> chip codenamed KF1950, so Kurt had to reverse-engineer it from scratch.

Reviewer 3: It might be good to describe a bit more here *why* this choice of chip (and what is the KF1950? Is it from MicroBT's WhatsMiner chips)?

And, in lines 168-170 below, describe *how* "opening up a chip" leads to the next sentence of "What it can do is let more people understand, build, repair, and take part in the hardware side of the Bitcoin community", i.e., it's quite a jump, perhaps link the two in a real-world, concrete way.

Reviewer: I made suggestion. Great that you pointed out this.

> chip codenamed KF1950, so Kurt had to reverse-engineer it from scratch.

Reviewer: ```suggestion
chip codenamed KF1950, so Kurt had to reverse-engineer it from scratch. Kurt chose this chip, made by MicroBT's WhatsMiner lineage, because they are sold for "virtually pennies", as Kurt puts is. The other more common chip, code name BM1370 by Bitmain, used by other open-source miners, costs about fifty dollars. 
```

>*But truly, the proudest moments were when I confirmed that the first Bitshoka
>hardware could communicate with the ASIC and later when I interfaced with a
>Whatsminer hashboard via the bitLode USB adapter board. Those initial "Hello
>ASIC" moments were amazing\!*

Opening up a chip like this will not make one tiny machine rival an industrial
mine. What it can do is let more people understand, build, repair, and take part

> in the hardware side of the Bitcoin community.

Reviewer: ```suggestion
Opening up a chip like this will not make one tiny machine rival an industrial
mine, and it doesn't have to. What it can do is let more people understand, build, repair, and take part
in the hardware side of the Bitcoin community.
```

>harder for a myriad of reasons.*

For anyone who wants to help, Kurt points to the [OSMU Discord][osmu-discord].
People can also message him there directly as @kuenrg. He's happy to help people
contributing to open-source bitcoin hardware. He's also apparently eager to give
monologues on open-source peer-to-peer energy.

Right now, Kurt is guiding the development of *Jua Kali*—"hot sun" in
Swahili—bitcoin miners at *Gridless*. He's also working on something he calls
*Stimatofali*—"stima" for electricity, "matofali" for bricks—an open-source way
for solar home systems to share power with each other, like snapping electric
bricks together.

Receiving the grant is what allowed Kurt to work on reverse engineering the
hardware full time. He says that without OpenSats, he wouldn't have had the time
or justification to spend months studying the boards, translating the knowledge
into a schematic, and then creating the circuit board.

>*OpenSats has provided incredible support. \[...\] The support from OpenSats
>helped me during every challenge along the way; without them, I would have felt
>that I should probably be doing my time and energy elsewhere.*

> So, for now, he waits. The new board is ordered, somewhere across Nairobi. The

Reviewer 3: Thoughts when I read this: *It's a shame he has to wait! How long until the new board comes? Maybe order several to have some extra in stock?*

Reviewer: Hahaha, that's a great thought! He actually did order 10 more PCBAs (because he has to, but also he's an engineer, so that's his thing :D )

> logic analyzer stays out, ready.

Reviewer: ```suggestion
So, for now, Kurt waits. He ordered ten new boards that are being made in the Nairobi fab, according to instructions he created. Until then, his logic analyzer and soldering iron stays out, ready. 
```

Sometimes, against the odds, one of those small open-source miners scattered
across the world [finds a block][finds-a-block]—not bad for underdogs working
from home. "When I look up from my work and the mess in front of me," he says,
"I see a small lawn and a bunch of banana trees. There are often a bunch of
bananas on the trees, which we try to harvest before the monkeys get them."
Kurt, who hacked a computer chip and kept a fifty-year-old power supply alive,
likes his odds.

[bitshoka-nini]: https://github.com/kuenrg153/BitshokaNini_V1.2
[kuenrg-github]: https://github.com/kuenrg153
[open-hardware]: https://opensats.org/blog/open-hardware-for-open-money

> [bitaxe]: https://github.com/skot/bitaxe

Reviewer: ```suggestion
[bitaxe]: https://opensats.org/projects/bitaxe
```

Reviewer 2: Awesome, just wanted to make that change myself.

And this is where [open hardware][open-hardware] starts to matter: if the miners
are expensive to import, hard to repair, and built by a handful of companies far
away, then the energy may be local, but the power still isn't.

[*Bitshoka Nini*][bitshoka-nini] starts where most open-source projects do:
building upon existing work. "Shoka" in Swahili means axe, hence the
[*Bitaxe*][bitaxe], an open-source Bitcoin miner that was also funded by
OpenSats. In open source, copying is the point: someone publishes the files,
then somebody else picks them up, builds, breaks, and fixes them before putting
the work back into the wild.

The main difference between the two is the ASIC, or the application-specific
integrated circuit. An ASIC is a specialist in a world of generalists: useless
at nearly everything, but frighteningly good at one thing. Namely, guessing
numbers until you get the right one. However, no one had publicly analyzed the
chip codenamed KF1950, so Kurt had to reverse-engineer it from scratch. Kurt
chose this chip, made by MicroBT's WhatsMiner lineage, because they are sold for
"virtually pennies", as Kurt puts is. The other more common chip, code name

> BM1370 by Bitmain, used by other open-source miners, costs about fifty dollars. 

Reviewer: ```suggestion
chip codenamed KF1950, so Kurt had to reverse-engineer it from scratch. Kurt
chose this chip, made by MicroBT for their WhatsMiner machines, because they are sold for "virtually pennies", as Kurt puts it. The other more common chip, Bitmain's BM1370, used by other open-source miners, costs about fifty dollars or so. 
```

---
title: "Made in Africa: Kurt and the Open-Source Bitcoin Miner That Anyone Can Make"
date: '2026-05-22'
tags: ['OpenSats', 'bitcoin', 'hardware', 'spotlight']
authors: ['ville']
images: ['/static/images/spotlight/kurt-unger/featured.jpg', '/static/images/spotlight/kurt-unger/hero.jpg']
draft: false

> summary: "A developer spotlight on Kurt Unger, an OpenSats grantee who's building Bitcoin mining hardware that's understandable, repairable, and open."

Reviewer: ```suggestion
summary: "A developer spotlight on Kurt Unger, an OpenSats grantee who's building locally a Bitcoin mining hardware that's understandable, repairable, and open."
```

Reviewer: Actually, maybe not,... I mean it's obvious Kurt is doing it locally. Where else?

member answered immediately: "Vires in Numeris," a Latin phrase meaning
"Strength in Numbers." The saying originates all the way back to 2011 in the
[BitcoinTalk][bitcointalk] internet forum, when people still experimented with
the money.

The argument is not that one home miner can outmuscle an industrial operation.
It is that many small miners, spread across many places, make the system harder
to pressure through a few large targets.

---

And then *Nini* failed.

It wasn't the Nairobi dust, but rather the board could talk to the chip, but it
couldn't mine. Kurt traced the issue to the power circuit and redesigned the
board. He ordered a new version, but it hasn't arrived yet.

Despite the setback, Kurt says that he is able to keep going "because of support
from my wife (and kids) as well as the great community \[he has\] in
Nairobi—both in the Bitcoin space and out of it." He continues, "seeing the work
that others around me are doing, and how my work can support them and vice
versa, is also very encouraging."

> Hardware development is hard; hardware development in Africa is even
> harder for a myriad of reasons.

> For anyone who wants to help, Kurt points to the [OSMU Discord][osmu-discord].

Reviewer: ```suggestion
For anyone who wants to help, Kurt points to the OSMU Discord.
```

Reviewer: Deleted the osmu discord links, because they're stale and in osmu discord they said it's up to something called discord boosts, and i'm not sure they are going to have that custom link any time soon.

Sometimes, against the odds, one of those small open-source miners scattered
across the world [finds a block][finds-a-block]—not bad for underdogs working
from home. "When I look up from my work and the mess in front of me," he says,
"I see a small lawn and a bunch of banana trees. There are often a bunch of
bananas on the trees, which we try to harvest before the monkeys get them."
Kurt, who hacked a computer chip and kept a fifty-year-old power supply alive,
likes his odds.

---

This is the first post of our [OpenSats spotlight] series. For comments,
corrections, or suggestions, please reach out to spotlight@opensats.org. If you
want to support Kurt directly, join the [OSMU community][osmu] and start
building. Our support for Kurt's work was made possible thanks to your generous
donations to the [General Fund].

[bitshoka-grant]: /blog/eleventh-wave-of-bitcoin-grants#bitshoka
[bitshoka-nini]: https://github.com/kuenrg153/BitshokaNini_V1.2
[kuenrg-github]: https://github.com/kuenrg153
[open-hardware]: /blog/open-hardware-for-open-money
[bitaxe]: /projects/bitaxe
[osmu]: https://osmu.wiki/
[bitcointalk]: https://bitcointalk.org/index.php?topic=4994.msg140770#msg140770

> [osmu-discord]: https://discord.gg/osmu

Reviewer: ```suggestion
```
